Client/src/main/java/org/projectcardboard/client/view/battleview/PlayerBox.java

```diff
@@ -220,13 +222,34 @@ private void updateMP(int maxMP) {
                 e.printStackTrace();
             }
         }
+
         Text player2MPText = new Text(player2.getCurrentMP() + "/" + maxMP);
         player2MPText.setX(SCREEN_WIDTH - SCALE * (335 + maxMP * 40));
         player2MPText.setY(SCALE * (180 - maxMP * 2));
         player2MPText.setFont(Constants.AP_FONT);
         player2MPText.setStyle("-fx-text-base-color: white; -fx-font-size: 24px;");
         player2MPText.setFill(Color.AQUA);
         mpGroup.getChildren().add(player2MPText);
+        handleOpponentDeckView(maxMP);
+
+    }
+
+    public void handleOpponentDeckView(int maxMP){
+        Text opponentDeckInfo = new Text("");
+        if(battleScene.getMyPlayerNumber() == player1.getPlayerNumber()){
+            opponentDeckInfo.setText("Deck: " + player2.getDeckSize() + " Hand: " + player2.getHand().size() + "/6");
```

Comment:
"/6"  shouldn't be hard-coded.  use MAX_HAND_SIZE constant

Reply:
fixed